### DISCUSSION TITLE
How do I lazily load from base

### DISCUSSION THREAD

**Discussion by YusufSuleman** [pocketbase/pocketbase#1941] [Q&A] [answered]
Hello, I am using PocketBase to create a blog website (For fun). I can load from posts like this:
```ts
    return await pb.collection('posts').getList(2, 10, {
        sort: "-created,id"
    });
```
How would I load slowly from there, what signal should I give PocketBase, if I scroll down. I am using SolidJS. To not have a large  amount of unused data.

**Comment by ganigeorgiev** [comment, marked as answer]
You have 2 options for pagination:

1. You can increment the first argument of `getList(i, 10, {...})` reacting to each "scroll-end" event accordingly.

2. Use a datetime field value for cursor-based pagination, and pass the last records field value as filter on each "scroll-end" event, something like:
   ```js
   await pb.collection("example").getList(1, 10, {
       sort: "created",
       filter: `created > "${lastCreatedVal}"` // the initial lastCreatedVal could be empty string
   })
   ```


How exactly you would implement infinite scroll with SolidJS I'm not sure. You can try searching for an external library or try to implement it on your own for example using [`IntersectionObserver`](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) or other similar browser apis.

**Reply by YusufSuleman** [reply]
I can't go through record[0].items, it uses the fallback, It should be an array. Here is code for context:
```tsx
                    <div id="posts">
                <For fallback="Failed" each={records[0].items}>{(record:any, i) =>
                        <div>Posts
                            {record.title}
                        </div>
                        
                    }</For>
                    </div>
```

**Reply by ganigeorgiev** [reply]
I don't understand the above code sample.

What is `records`? If `records` in the above is the result of `getList()` then the way you are accessing the items with an index is incorrect. The available result fields are - https://github.com/pocketbase/js-sdk/blob/master/src/models/utils/ListResult.ts.

Could you provide a more detailed code sample? It is hard to understand what is going on without context.

**Reply by YusufSuleman** [reply]
Here is the whole example:
```tsx
import Sidebar from "./Sidebar"
import PocketBase from 'pocketbase';
import { createEffect, createResource, createSignal, For } from "solid-js";
import { useParams } from "@solidjs/router";


function Blog(_props:any) {
    const params = useParams();
    let pageNumber:any = params.pageNumber

    async function getPosts(_props:any) {
        const pb = new PocketBase('http://127.0.0.1:8090');
        return await pb.collection('posts').getList(pageNumber, 30, {
            sort: "-created,id"
        });
    }

    
    const records:any = createResource(getPosts)
    createEffect(() => {
        console.log("all",records)
    })
    console.log(records)
    return (
        <div class="content-body">
                <div class="post">
                    <div class="title">
                    Lol
                    <div id="posts">
                <For fallback="Failed" each={records[0].items}>{(record:any, i) =>
                        <div>Posts
                            {record.title}
                        </div>
                        
                    }</For>
                    </div>
                </div>
            </div>
            
           <Sidebar></Sidebar> 
        </div>
    )
}

export default Blog
```

**Reply by ganigeorgiev** [reply]
What is the output of the `console.log`? Is it an object with `{ page, perPage, totalItems, totalPages, items }` keys?

Per the [SolidJS createResource docs](https://www.solidjs.com/docs/latest#createresource) and the example in the official ["Async/Resources" tutorial](https://www.solidjs.com/tutorial/async_resources?solved), the first return argument of the `createResource` is a "read" function, aka. meaning that you have to call `records[0]().*` before accessing the result (at least once; see the example in the tutorial they are calling `user()` inside the `JSON.stringify`).

Or in other words, you can try something like this:

```jsx
<For fallback="Failed" each={records[0]()?.items}>{(record, i) =>
    <div>
        Posts {record.title}
    </div>
}</For>
```

Note that we are accessing the items with `?` because the `fetch` may still be loading.

I'd advice to rename your `records` variable to something else to avoid the confusion. Alternatively you can also use array destructuring assignment as shown in the SolidJS documentation and skip the `[0]` index access altogether, aka.:

```jsx
const [result] = createResource(getPosts)

...


<For fallback="Failed" each={result()?.items}>{(record, i) =>
    <div>
        Posts {record.title}
    </div>
}</For>
```

**Reply by ganigeorgiev** [reply]
Additionally, also note that if PocketBase returns empty `items` array, then make sure that your client request satisfies the collection List/Search API rule (you can leave the rule empty to allow access to everyone).

**Reply by YusufSuleman** [reply]
Thanks, now it shows all the titles. Now I have to make actual post previews.
